src/hp/magicsock/derp_actor.rs

```diff
@@ -0,0 +1,810 @@
+use std::{
+    collections::{hash_map, HashMap, HashSet},
+    net::IpAddr,
+    sync::{atomic::Ordering, Arc},
+    time::{Duration, Instant},
+};
+
+use backoff::backoff::Backoff;
+use bytes::{Bytes, BytesMut};
+use tokio::{sync::mpsc, time};
+use tokio_util::sync::CancellationToken;
+use tracing::{debug, info, trace, warn};
+
+use crate::{
+    hp::{
+        derp::{self, DerpMap, MAX_PACKET_SIZE},
+        key::{self, node::PUBLIC_KEY_LENGTH},
+    },
+    inc,
+    metrics::magicsock::MagicsockMetrics,
+    record,
+};
+
+use super::conn::{ActorMessage, Inner};
+
+/// How long a non-home DERP connection needs to be idle (last written to) before we close it.
+const DERP_INACTIVE_CLEANUP_TIME: Duration = Duration::from_secs(60);
+
+/// How often `clean_stale_derp` runs when there are potentially-stale DERP connections to close.
+const DERP_CLEAN_STALE_INTERVAL: Duration = Duration::from_secs(15);
+
+pub(super) enum DerpActorMessage {
+    Send {
```

Comment:
Can we have small comments about each enum case? I mean in case of `Send` it is self explaning. But for e.g. `NotePreferred` or `MaybeCloseDerpsOnRebind` a bit of text help.